api/institutions/views.py

```diff
@@ -420,3 +421,43 @@ def get_default_queryset(self):
     # overrides RetrieveAPIView
     def get_queryset(self):
         return self.get_queryset_from_request()
+
+class InstitutionUserMetricsList(JSONAPIBaseView, ListFilterMixin, generics.ListAPIView, InstitutionMixin):
+
+    permission_classes = {
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        IsInstitutionalMetricsUser,
+    }
+
+    required_read_scopes = [CoreScopes.INSTITUTION_METRICS_READ]
+    required_write_scopes = [CoreScopes.NULL]
+
+    serializer_class = InstitutionUserMetricsSerializer
+
+    view_category = 'institutions'
+    view_name = 'institution-user-metrics'
+
+    def get_default_queryset(self):
+        institution = self.get_institution()
+        queryset = MockQueryset()
+
+        results = UserInstitutionProjectCounts.get_current_user_metrics(institution)
+        for user_record in results:
+            record_dict = {}
+            user_id = user_record.user_id
+            fullname = OSFUser.objects.get(guids___id=user_id).fullname
+            record_dict['user'] = f'({user_id}) {fullname}'
+            record_dict['user_id'] = user_id
+            record_dict['public_projects'] = user_record.public_project_count
+            record_dict['private_projects'] = user_record.private_project_count
+            record_dict['department'] = user_record.department
+            record_dict['id'] = user_record.institution_id
+
+            queryset.add_dict_as_item(record_dict)
```

Comment:
Out of curiosity, did you try just returning `results` instead of doing this for loop, and then using `source`s in the serializer to change the field name?

Reply:
If that doesn't work, I think you could still just `queryset.add_dict_as_item(user_record)` after putting the `user` on the `record_dict` and save yourself some work.